qubesbuilder/plugins/source_deb/__init__.py

```diff
@@ -236,18 +236,30 @@ def run(self, stage: str):
             ]
 
             if package_type == "quilt":
-                # Create archive if no external file is provided.
-                if not self.parameters.get("files", []):
+                if self.component.is_salt():
+                    copy_in += [
+                        (self.plugins_dir / "source/salt/Makefile.install", source_dir)
+                    ]
+                    cmd += [
+                        f"{PLUGINS_DIR}/source/salt/yaml-dumper "
+                        "--env VERBOSE "
+                        f"--outfile {source_dir}/Makefile.vars -- "
+                        f"{PLUGINS_DIR}/source/salt/FORMULA-DEFAULTS {source_dir}/FORMULA"
+                    ]
+
+                # Create archive only if no external files are provided or if explicitly requested.
+                if not self.parameters.get("files", []) or self.parameters.get(
+                    "create-archive", False
```

Comment:
This doesn't allow forcing both ways (explicit `create-archive: false` will not prevent creating it). Maybe something like:
```python
create_archive = not self.parameters.get("files", [])
create_archive = self.parameters.get("create-archive", create_archive)
if create_archive:
    ...
```